File: src/main/java/com/example/webfluxdemo/adapters/UserAdapter.java
package com.example.webfluxdemo.adapters;

import com.example.webfluxdemo.AbstractAdapter;
import com.example.webfluxdemo.model.User;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

@Component
public class UserAdapter extends AbstractAdapter<User> {

    private static final String URL_PREFIX = "http://localhost:8040/api/users/";

    public UserAdapter() {
        this(new RestTemplate());
    }

    public UserAdapter(RestTemplate restTemplate) {
        super(restTemplate, URL_PREFIX);
    }

    public User getUser(int userId) {
        return sendRequest(Integer.toString(userId), User.class);
    }

}

File: src/main/java/com/example/webfluxdemo/adapters/UserAdapter.kt
package com.example.webfluxdemo.adapters

import com.example.webfluxdemo.AbstractAdapter
import com.example.webfluxdemo.model.User
import org.springframework.stereotype.Component
import org.springframework.web.reactive.function.client.WebClient

private const val URL_PREFIX: String = "http://localhost:8040/api/users/"

@Component
class UserAdapter(webClient: WebClient = WebClient.builder().build()) : AbstractAdapter(webClient, URL_PREFIX) {
    suspend fun getUser(userId: Int): User =
            sendRequest(userId.toString())
}